.github/workflows/sel4test-hw.yml

```diff
@@ -1,113 +1,26 @@
 
 jobs:
-  hw-build:
-    name: HW Build
-    runs-on: ubuntu-latest
-    if: ${{ github.event_name == 'push' ||
-            github.event_name == 'pull_request_target' &&
-              github.event.action != 'labeled' &&
-              (contains(github.event.pull_request.labels.*.name, 'hw-build') ||
-               contains(github.event.pull_request.labels.*.name, 'hw-test')) ||
-            github.event_name == 'pull_request_target' &&
-              github.event.action == 'labeled' &&
-              (github.event.label.name == 'hw-build' ||
-               github.event.label.name == 'hw-test') }}
-    strategy:
-      fail-fast: false
-      matrix:
-        # There is no "rv32imac" hardware yet.
-        march: [armv7a, armv8a, nehalem, rv64imac]
-        compiler: [gcc, clang]
-    steps:
-    - name: Build
-      uses: seL4/ci-actions/sel4test-hw@master
-      with:
-        march: ${{ matrix.march }}
-        compiler: ${{ matrix.compiler }}
-        sha: ${{ github.event.pull_request.head.sha }}
-    - name: Upload images
-      uses: actions/upload-artifact@v4
-      with:
-        name: images-${{ matrix.march }}-${{ matrix.compiler }}
-        path: '*-images.tar.gz'
-    - name: Upload kernel.elf files
-      uses: actions/upload-artifact@v4
-      with:
-        name: kernel.elf-${{ matrix.march }}-${{ matrix.compiler }}
-        path: '*-kernel.elf'
-
-  the_matrix:
-    name: Matrix
-    needs: hw-build
-    runs-on: ubuntu-latest
-    outputs:
-      matrix: ${{ steps.matrix.outputs.matrix }}
-    steps:
-    - id: matrix
-      uses: seL4/ci-actions/sel4test-hw-matrix@master
-
-  hw-run:
-    name: HW Run
-    runs-on: ubuntu-latest
-    needs: the_matrix
-    if: ${{ github.repository_owner == 'seL4' &&
-            (github.event_name == 'push' ||
-             github.event_name == 'pull_request_target' &&
-               github.event.action != 'labeled' &&
-               contains(github.event.pull_request.labels.*.name, 'hw-test') ||
-             github.event_name == 'pull_request_target' &&
-               github.event.action == 'labeled' &&
-               github.event.label.name == 'hw-test') }}
-    strategy:
-      fail-fast: false
-      matrix: ${{ fromJson(needs.the_matrix.outputs.matrix) }}
-    steps:
-      - name: Get machine queue
-        uses: actions/checkout@v4
-        with:
-          repository: seL4/machine_queue
-          path: machine_queue
-      - name: Download image
-        uses: actions/download-artifact@v4
-        with:
-          name: images-${{ matrix.march }}-${{ matrix.compiler }}
-      - name: Run
-        uses: seL4/ci-actions/sel4test-hw-run@master
-        with:
-          platform: ${{ matrix.platform }}
-          compiler: ${{ matrix.compiler }}
-          mode: ${{ matrix.mode }}
-          index: $${{ strategy.job-index }}
-        env:
-          HW_SSH: ${{ secrets.HW_SSH }}
+  sel4test:
+    name: seL4Test
+    uses: seL4/ci-actions/.github/workflows/sel4test-hw.yml@master
```

Comment:
I wonder if we should keep the label filtering here, as this give a better control when the called workflow runs at all. Also, we could pass it parameters that allow controlling the behavior:
```
    if: ${{ ((github.event.action == 'labeled') &&
             (github.event.label.name == 'hw-bench')) ||
            ((github.event.action != 'labeled') &&
             contains(github.event.pull_request.labels.*.name, 'hw-bench')) }}
    uses: axel-h/ci-actions/.github/workflows/sel4bench-hw.yml@patch-axel-7
    with:
      context: pr-${{ github.event.number }}
      labels: join(github.event.pull_request.*.name, ', ')
```
Then the called workflow could simply use globla concurrency

Reply:
No, the whole point is to get all of this out of this repo and use one central workflow in as many other repos as possible. If we have to make the central workflow a bit more complex, that is completely fine. Updating 20 repos each time we want to make a change is not fine, that's why those workflows are currently missing in most repos.